README.md

```diff
@@ -140,25 +141,35 @@ In the diagram below, portions of Flow X and Flow Y are combined to form Flow Z,
 
 In practice a client may need to read (and potentially decode, in the case of inter-frame video codecs) the entire object, discarding the unwanted grains and returning those selected by the segment to the consumer.
 This can be handled by mapping the object's timeline into the Flow timeline by adding `ts_offset` to each timestamp inside the media object, and then discarding grains when the resulting timestamp falls outside the given `timerange` for the segment.
-Alternatively if `sample_offset` and `sample_count` are set on the segment, a client may count the samples read from the object and discard accordingly, although care must be taken with temporal re-ordering, since the `sample_offset` and `sample_count` refer to the presentation timeline, rather than the decode timeline.
+Alternatively if `sample_offset` and `sample_count` are set on the segment, a client may count the samples read from the object and discard accordingly, although care must be taken with codecs that use temporal re-ordering, since the `sample_offset` and `sample_count` refer to the presentation timeline, rather than the decode timeline.
 
 ![Graphic showing the objects making up Flow Z (above) and their segments, along with the selected grains](./docs/images/Flow%20and%20Media%20Timelines-Flow%20XYZ-subsegments.drawio.png)
 
-The diagram above shows three of the objects used by Flow Z and a smaller portion of the Flow Z timeline considered previously, with both the media object timeline (and `media_ts`) and Flow timeline (`grain_ts`) of each grain in the object along the bottom.
+The diagram above shows three of the objects used by Flow Z and a smaller portion of the Flow Z timeline considered previously, with both the media object timeline (and `media_ts`) and Flow timeline (`segment_ts`) of each grain in the object along the bottom.
 The `timerange`/`ts_offset` approach is illustrated by the following pseudocode, which shows how a client might apply the `ts_offset` to each `media_ts`, then validate whether it is inside the given `timerange`.
 Note that the `media_ts` may be a different precision or timebase to the nanosecond timestamps used by TAMS, so some conversion may be required.
 
 ```python
-grain_ts = media_ts_to_timestamp(media_ts)         # `4:0` for `media_ts = 4.0`
-grain_ts_in_flow_timeline = grain_ts + ts_offset   # `4:0 + -0:700000000 = 3:300000000`
-if (grain_ts_in_flow_timeline < timerange.start or
-    grain_ts_in_flow_timeline >= timerange.end):   # `3:300000000 is less than 3:500000000`
-    discard_grain()                                # So this grain is discarded
+sample_ts = ts_to_timestamp(ts)      # `4:0` for `ts = 4.0sec`
```

Comment:
NIT: We commonly use `ts` as a shorthand for TAMS `timestamp`s. We're using that here, in fact, in the naming of variables. So maybe `ts_to_timestamp(ts)` is a little confusing? But I guess it is consistent with ffmpeg :upside_down_face: